feat(imagen): add image modification support with text prompts

TimeToBuildBob opened the pull request:
## Summary

Implements Issue #77: Add image + text modification capability to imagen plugin.

## Changes

### New Function: `modify_image()`

Enables users to modify existing images with text prompts, enabling iterative design workflows.

```python
from gptme_image_gen.tools.image_gen import modify_image

# Change background
result = modify_image(
    image_path='avatar.png',
    prompt='change the background to a sunset beach scene',
    view=True
)

# Add accessories
result = modify_image(
    image_path='portrait.png',
    prompt='add round glasses and a blue hat',
    output_path='portrait_with_accessories.png'
)
```

### Features
- Uses Gemini's multimodal API to accept both image and text input
- Supports `view` parameter to display results to LLM
- Includes cost tracking for modification operations
- Comprehensive test coverage

### Limitations
- Currently only Gemini supports image modification with text prompts
- For DALL-E users can use `generate_variation()` or `generate_image()` with detailed prompts

## Use Cases
- Avatar creation and refinement
- Iterative design work
- Fine-tuning generated images
- A/B testing small variations

## Testing
- Added 6 unit tests for the new functionality
- All tests pass locally

Closes #77

---
Co-authored-by: Bob <bob@superuserlabs.org>
<!-- ELLIPSIS_HIDDEN -->


----

> [!IMPORTANT]
> Adds `modify_image()` to `imagen` plugin for text-based image modifications using Gemini's API, with comprehensive test coverage.
> 
>   - **New Functionality**:
>     - Adds `modify_image()` in `image_gen.py` to modify images using text prompts.
>     - Supports iterative design workflows with text-based image modifications.
>     - Utilizes Gemini's multimodal API for image and text input.
>   - **Features**:
>     - Supports `view` parameter to display results to LLM.
>     - Includes cost tracking for modification operations.
>     - Comprehensive test coverage added in `test_image_gen_features.py` and `test_image_gen_phase2.py`.
>   - **Limitations**:
>     - Only Gemini supports image modification with text prompts.
>     - DALL-E users can use `generate_variation()` or `generate_image()` with detailed prompts.
>   - **Testing**:
>     - Added 6 unit tests for new functionality.
>     - All tests pass locally.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=gptme%2Fgptme-contrib&utm_source=github&utm_medium=referral)<sup> for 6d6c10eb8df0fe7b47b0cff2df832ce82e4ebe6a. You can [customize](https://app.ellipsis.dev/gptme/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>


<!-- ELLIPSIS_HIDDEN -->

ErikBjare commented:
Hmm, I think we should perhaps review this design. Look up the actual capabilities of e.g. Nano Banana Pro (multiple images as input, e.g. to provide character/scene references)

The current API is maybe a good start for some models, but not general enough to capture the capabilities of Nano Banana (multiple images as input).

You might want to use Perplexity to look up the API capabilities and then adjust the design.

TimeToBuildBob commented:
## Research: Nano Banana Pro / Gemini Multi-Image Capabilities

Thanks for the design review feedback, @ErikBjare! I researched the API capabilities.

### Key Findings

**Gemini 2.x/3.x Multi-Image Support**:
- Can accept **up to 14 images** as input via `contents=[prompt, img1, img2, ...]`
- Enables:
  - Character/scene references from multiple images
  - Style blending (e.g., "keep character from img1, background mood from img2")
  - Multi-reference integration (combine people, clothing, etc.)

**Example API Usage** (from Google SDK):
```python
from google import genai
from PIL import Image

client = genai.Client(api_key="...")
img1 = Image.open("ref1.png")
img2 = Image.open("ref2.jpg")

response = client.models.generate_content(
    model="gemini-2.5-flash-image",
    contents=[prompt, img1, img2],  # Multiple images!
)
```

### Design Options

**Option A**: Extend `modify_image()` - Minimal Change
```python
def modify_image(
    image_path: str | list[str],  # Accept single or list
    prompt: str,
    ...
)
```
- ✅ Backward compatible
- ✅ Simple implementation
- ❌ Semantic mismatch: "modify" implies single image

**Option B**: New `composite_images()` Function
```python
def composite_images(
    reference_images: list[str],
    prompt: str,
    output_path: str | None = None,
    ...
) -> ImageResult
```
- ✅ Clear intent: multi-reference generation
- ✅ Clean API separation
- ❌ Two similar functions

**Option C**: General Reference Images API
```python
def generate_image(
    prompt: str,
    reference_images: list[str] | None = None,  # NEW param
    ...
)
```
- ✅ Most flexible - works for all modes
- ✅ Single API for all generation types
- ❌ Larger change to existing API

### Recommendation

I lean toward **Option B** (new `composite_images()` function) because:
1. Keeps `modify_image()` simple for single-image edits
2. Multi-image composition is a distinct use case
3. Avoids confusing semantics

However, Option C could be more elegant long-term if we want a unified API.

**What approach would you prefer?** Happy to implement whichever direction you choose.

TimeToBuildBob commented:
## Option C Implemented ✅

Thanks @ErikBjare for the design direction! I've implemented Option C - adding an `images` parameter to `generate_image()`.

### Changes

**New `images` parameter:**
```python
def generate_image(
    prompt: str,
    images: list[str] | str | None = None,  # NEW
    ...
)
```

**Usage examples:**
```python
# Generate from text only (existing behavior)
result = generate_image('a sunset over mountains')

# Modify a single image
result = generate_image(
    prompt='change the background to a beach',
    images='portrait.png'
)

# Multi-reference generation (character + style references)
result = generate_image(
    prompt='create a portrait in this style with this character',
    images=['character_ref.png', 'style_ref.png']
)
```

### Technical Details
- Parameter named `images` (not 'reference_images') as requested
- Docstring explains various use cases (modification, references, combining elements)
- Validates images exist before generation
- Only Gemini supports multi-image input (DALL-E raises clear error)
- `modify_image()` retained for backward compatibility

### Tests
All 90 tests pass. Updated existing tests to work with new signature.

### What's Next?
- Consider if `modify_image()` should be deprecated in favor of unified API
- Could add tests specifically for multi-image generation scenarios

Ready for review!

TimeToBuildBob commented:
## ✅ Removed `modify_image()` per request

@ErikBjare - Removed `modify_image()` function as requested since we now have the unified `images` parameter in `generate_image()`.

### Changes
- Removed `modify_image()` function and `_modify_gemini()` helper
- Deleted `test_image_modify.py` test file (6 tests)
- Updated tool instruction examples to use `generate_image(images=...)` instead
- Removed `modify_image` from exported functions list

### API for modifications is now:
```python
# Single image modification
generate_image(
    prompt='change background to sunset',
    images='portrait.png'
)

# Multi-reference generation  
generate_image(
    prompt='create portrait in this style with this character',
    images=['character_ref.png', 'style_ref.png']
)
```

All 84 remaining tests pass. Ready for final review!
